webpack.config.js

```diff
@@ -0,0 +1,100 @@
+const path = require('path');
+const webpack = require('webpack');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const FaviconsWebpackPlugin = require('favicons-webpack-plugin');
+const production = process.env.NODE_ENV === 'production';
+
+console.log('*** Building production?', production);
+
+const port = 3000;
+const publicPath = '/';
+
+const config = {
+    mode: production ? 'production' : 'development',
+    entry: './src/app/main.js',
+    output: {
+        chunkFilename: 'chunk-[chunkhash].js',
+        filename: production ? '[name]-[hash].min.js' : '[name].js',
+        path: path.resolve(__dirname, 'dist'),
+        publicPath
+    },
+    devtool: production ? 'source-map' : 'inline-source-map',
+    module: {
+        rules: [
+            { test: /\.jsx?$/, exclude: /node_modules/, loader: 'babel-loader' },
+            {
+                test: /\.scss$/,
+                use: [ 'style-loader', 'css-loader',
+                    {
+                        loader:  'fast-sass-loader',
+                        options: {
+                            implementation: require('node-sass'),
+                            includePaths: [ './src/styles', './node_modules' ]
+                        }
+                    }
+                ]
+            }
+        ]
+    },
+    resolve: {
+        alias: {
+            'react': 'preact/compat',
+            'react-dom/test-utils': 'preact/test-utils',
+            'react-dom': 'preact/compat',
+            '~': path.resolve(__dirname, 'src/app'),
+        },
+        modules: [
+            path.resolve(__dirname, 'src/app'),
+            'node_modules',
+        ],
+        extensions: ['.js', '.jsx']
+    },
+    optimization: {
+      splitChunks: {
+        chunks: 'all',
+        maxInitialRequests: production ? 5 : 1,
+      },
+    },
+    plugins: [
+        new HtmlWebpackPlugin({
+            template: './src/index.html'
+        }),
+        new FaviconsWebpackPlugin('./src/images/favicon.svg')
+    ],
+    performance: {
+      maxEntrypointSize: 2.5 * 1000000, // 1MB
+      maxAssetSize: 2.1 * 1000000,
+    },
+    watchOptions: {
+      aggregateTimeout: 500,
+      poll: 1000,
+    },
+    devServer: {
+      clientLogLevel: 'warning',
+      contentBase: path.join(__dirname, './src'),
+      contentBasePublicPath: '/',
+      filename: '[name].js',
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
+        'Access-Control-Allow-Headers': 'X-Requested-With, content-type, Authorization',
+      },
```

Comment:
you might not need CORS since you're using html web pack plugin.  I think all requests should be on the same host/port then.  Tutor needs it since the html is served on port 3001 by tutor backend, but the assets are on port 8001

Reply:
On dev (localhost), I pull assets from cms-dev. :)